minkindr_conversions/include/minkindr_conversions/kindr_msg.h

```diff
@@ -37,46 +39,91 @@ void quaternionMsgToKindr(
 
 // Also the Eigen implementation version of this.
 template <typename Scalar>
-void quaternionKindrToMsg(const Eigen::Quaternion<Scalar>& kindr,
-                          geometry_msgs::Quaternion* msg) {
+void quaternionKindrToMsg(
+    const Eigen::Quaternion<Scalar>& kindr, geometry_msgs::Quaternion* msg) {
   CHECK_NOTNULL(msg);
   quaternionEigenToMsg(kindr, *msg);
 }
 
 template <typename Scalar>
-void quaternionMsgToKindr(const geometry_msgs::Quaternion& msg,
-                          Eigen::Quaternion<Scalar>* kindr) {
+void quaternionMsgToKindr(
+    const geometry_msgs::Quaternion& msg, Eigen::Quaternion<Scalar>* kindr) {
   CHECK_NOTNULL(kindr);
   Eigen::Quaternion<double> kindr_double;
   quaternionMsgToEigen(msg, kindr_double);
   *kindr = kindr_double.cast<Scalar>();
 }
 
+template <typename Scalar>
+void rotationKindr2DToMsg(
+    const Eigen::Rotation2D<Scalar>& kindr, geometry_msgs::Quaternion* msg) {
+  CHECK_NOTNULL(msg);
+  const kindr::minimal::RotationQuaternionTemplate<Scalar> quat(
+      kindr::minimal::AngleAxisTemplate<Scalar>(
+          kindr.angle(), static_cast<Scalar>(0.0), static_cast<Scalar>(0.0),
+          static_cast<Scalar>(1.0)));
+  quaternionEigenToMsg(quat.toImplementation(), *msg);
+}
+
+template <typename Scalar>
+void quaternionMsgToKindr2D(
+    const geometry_msgs::Quaternion& msg, Eigen::Rotation2D<Scalar>* kindr) {
+  CHECK_NOTNULL(kindr);
+  CHECK_LT(std::abs(msg.x), std::numeric_limits<double>::epsilon())
+      << "No proper 2D rotation.";
+  CHECK_LT(std::abs(msg.y), std::numeric_limits<double>::epsilon())
+      << "No proper 2D rotation.";
+  kindr::minimal::RotationQuaternionTemplate<Scalar> quat;
+  quaternionMsgToEigen(msg, quat.toImplementation());
+  const kindr::minimal::AngleAxisTemplate<Scalar> angle_axis(quat);
```

Comment:
I think you need `quat.cast<Scalar>()`

Reply:
@dymczykm but isn't this already `Scalar`?

Reply:
nvmnd